smalltalksrc/Slang/CCodeGenerator.class.st

```diff
@@ -4397,6 +4415,38 @@ CCodeGenerator >> removeVariable: aName ifAbsent: ifAbsentBlock [
 	^variables remove: aName ifAbsent: ifAbsentBlock
 ]
 
+{ #category : #translating }
+CCodeGenerator >> renameKeywordsConflicts [
+
+	methods do: [ :method |
+		| localRewording cSelector |
+		"rename locals"
+		localRewording := Dictionary new.
+
+		(method allLocals select: [ :local |
+			 self reservedWords includes: local ]) do: [ :local |
+			localRewording at: local put: (self sanitizeKeyword: local) ].
+		localRewording ifNotEmpty: [
+			method renameVariablesUsing: localRewording ].
+
+		"translate selector"
+		cSelector := self cSelectorName: method selector.
+
+		(self reservedWords includes: cSelector) ifTrue: [
+			self
+				addSelectorTranslation: method selector
+				to: (self sanitizeKeyword: cSelector) ] ].
+
+	"translate structs instance variables"
+	structClasses ifNotNil: [ :classes |
```

Comment:
This can be nil? :/

Reply:
yep, but I just noticed that the same if is in the getter so I'll just use that.